lib/rubocop/cop/rails/action_order.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces consistent ordering of the standard rails controller
+      # actions.
+      #
+      # The cop is configurable and can enforce any ordering of the standard
+      # actions. All order methods are ignored.
+      #
+      # [source,yaml]
+      # ----
+      #  Rails/ActionOrder:
+      #    ExpectedOrder:
+      #      - index
+      #      - show
+      #      - new
+      #      - edit
+      #      - create
+      #      - update
+      #      - destroy
+      # ----
+      #
+      # @example
+      #   # bad
+      #   def index; end
+      #   end destroy; end
+      #   def show; end
+      #
+      #   # good
+      #   def index; end
+      #   def show; end
+      #   end destroy; end
+      class ActionOrder < Base
+        DEFAULT_ORDER = %w[index show new edit create update destroy].freeze
```

Comment:
The default order should be defined in `config/default.yml` using the same configuration someone would use to override it.